### PR TITLE
Vignette sensitivity analysis

### PR DESCRIPTION
**Pull request by PavelBal**

### CODE REVIEW

File: R/sensitivity-analysis-results.R [hunk 1]
```diff
@@ -57,6 +57,8 @@ SensitivityAnalysisResults <- R6::R6Class("SensitivityAnalysisResults",
     #' @param ... Rest arguments.
     print = function(...) {
       private$printClass()
```
Comment (PavelBal) [inline R/sensitivity-analysis-results.R:59]: I think it is a useful addition to the print method.

File: R/sensitivity-analysis.R [hunk 2]
```diff
@@ -26,7 +26,7 @@ SensitivityAnalysis <- R6::R6Class(
     #' @description
     #' Initialize a new instance of the class
     #' @param simulation Simulation for which a sensitivity analysis should be performed
-    #' @param parameterPaths List of parameters to use for sensitivity calculation (optional).If undefined, the sensitivity will be performed automatically
+    #' @param parameterPaths Vector of parameter paths to use for sensitivity calculation (optional).If undefined, the sensitivity will be performed automatically
```
Comment (PavelBal) [inline R/sensitivity-analysis.R:29]: Fixed the docu regarding list vs vector.

File: vignettes/create-run-population.Rmd [hunk 3]
```diff
@@ -29,7 +29,7 @@ myPopulation <- loadPopulation(csvPopulationFile = popFilePath)
 print(myPopulation)
 ```
 
-## Creating populations {#SecretionCreatePopulation}
```
Comment (PavelBal) [inline vignettes/create-run-population.Rmd:32]: Was a typo

File: vignettes/introduction-ospsuite.Rmd [hunk 4]
```diff
@@ -89,3 +90,9 @@ The most important classes are:
     : PK analyses for a simulations result.
   : `QuantityPKParameter`
     : A certain PK-parameter for an output quantity. Has the fields `$name` which is the name of the pk-parameter (e.g. "C_max"), `$quantityPath` the path of the output the parameter has been calculated for, and `$values` the value (or list of values for a population simulation).
+  : `SensitivityAnalysis`
```
Comment (PavelBal) [inline vignettes/introduction-ospsuite.Rmd:93]: Added class description to intro

File: vignettes/sensitivity-analysis.Rmd [hunk 5]
```diff
@@ -0,0 +1,118 @@
+---
```
Comment (PavelBal) [inline vignettes/sensitivity-analysis.Rmd:1]: Cover the basics of SA.

File: R/sensitivity-analysis-results.R (hunk 1 above)
Comment (msevestre) [reply R/sensitivity-analysis-results.R:59]: ok

File: vignettes/create-run-population.Rmd (hunk 3 above)
Comment (msevestre) [reply vignettes/create-run-population.Rmd:32]: lol

Comment (PavelBal) [reply vignettes/create-run-population.Rmd:32]: Too much modeling... :D

File: vignettes/sensitivity-analysis.Rmd [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+)
+```
+
+The models built with OSPS depend on a lot of input parameters which are based on literature values, measurements, databases, assumptions. For a given set of input parameters a number of output curves is computed in a simulation. To assess which input parameters have most impact on the output curves, a sensitivity analysis of the simulaiton can be performed. For more information about theoretical background, see [OSPS documentation](https://docs.open-systems-pharmacology.org/shared-tools-and-example-workflows/sensitivity-analysis).
+
+In brief, the values of the chosen parameters are changed by a certain percentage and the impact of these changes on PK parameters of model outputs is assessed.
+
+## Performing a sensitivity analysis
+The first step of performing a sensitivity analysis is creating an object of the class `SensitivityAnalysis`. At this step, the user can define which parameters should be considered for the sensitivity analysis, in which range the values are variated, and how may steps are performed in one direction (plus and minus). If no parameters are specified, all constant parameters of the simulation will be variated.
+
```
Comment (Yuri05) [inline vignettes/sensitivity-analysis.Rmd:24]: If no parameters are specified, all constant **and suitable for sensitivity calculations** parameters of the simulation will be variated.

"suitable for sensitivity calculations" means: 
- is not zero
- is **effectively** used in simulation (btw: is this parameter property accessible from R and if yes: is it also explained somewhere?)
- is non-categorical 
- is not marked with "CanBeVaried=false" in the simulation
- is not marked as read-only
- is not a table
@msevestre Anything else?
The list of all such parameters can be get from the simulation by `potentialVariableParameterPathsFor(simulation)`.

File: vignettes/sensitivity-analysis.Rmd [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+                                                                      "Aciclovir|Fraction unbound (plasma)",
+                                                                      "Organism|Age"))
+print(sa)
+
+# Show which parameters will be variated
+sa$parameterPaths
+```
+
+New parameters can be added to an existing `SensitivityAnalysis` by calling the method `addParameterPaths`. WARNING: If no parameters were specified during creating of a `SensitivityAnalysis` (all constant parameters are considered), calling `addParameterPaths` will make only the manually added parameters being variated.
+
```
Comment (Yuri05) [inline vignettes/sensitivity-analysis.Rmd:47]: same: all constant and suitable for sensi calculation

File: vignettes/sensitivity-analysis.Rmd [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+# Create a `SensitivityAnalysis` with all constant parameters
+sa <- SensitivityAnalysis$new(simulation = sim)
+
+# Run the sensitivity analysis
+saResult <- runSensitivityAnalysis(sa)
+print(saResult)
+```
+The method `allPKParameterSensitivitiesFor` returns a list of `PKParameterSensitivity` objets. `PKParameterSensitivity` describes the sensitivity (field `$value`) of a PK-Parameter (`$pkParameterName`) for the output `$outputPath` calculated for the variated parameter `$parameterPath`. The argument `totalSensitivityThreshold` of the method `allPKParameterSensitivitiesFor` is used to filter out the most impactful parameters. A threshold of 0.9 means that only parameters participating to a total of 90 percent of the sensitivity would be returned. A value of 1 would return the sensitivity for all parameters.
+
```
Comment (Yuri05) [inline vignettes/sensitivity-analysis.Rmd:81]: Is the threshold optional or mandatory in R? If optional: you could mention that 0.9 is also the **default** value

Comment (PavelBal) [reply vignettes/sensitivity-analysis.Rmd:81]: The threshold is optional. I would prefer implementing a method to read the variables of the ospsuiteEnv, where such defaults are defined, as proposed here https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/386

Then we could put a referenrece to it in the documentation.

Comment (msevestre) [reply vignettes/sensitivity-analysis.Rmd:81]: I agree I think we need this

**Pull request merged**